tools/makejwt/main.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/asn1"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+)
+
+var (
+	// ErrorEmptyPEMKey when the PEM key was not passed in
+	ErrorEmptyPEMKey = fmt.Errorf("empty PEM key or filename")
+
+	// ErrorEmptyAppId when the App ID was not passed in
+	ErrorEmptyAppId = fmt.Errorf("empty App Id")
+)
+
+type JWTSpec struct {
+	Base64PEMBytes string
+	AppID          string
+	Duration       time.Duration
+	Raw            bool
+}
+
+type jsonReply struct {
+	PublicKey string
+	JWT       string
+}
+
+func makeJWT(spec *JWTSpec) (string, error) {
+	// decode base64 of the key to PEM
+	pemBytes, err := base64.StdEncoding.DecodeString(string(spec.Base64PEMBytes))
+	if err != nil {
+		return "", fmt.Errorf("error decoding base64 of the PEM key: %v", err)
+	}
+
+	// Parse PEM block
+	block, _ := pem.Decode(pemBytes)
+	if block == nil || block.Type != "RSA PRIVATE KEY" {
```

Comment:
let's not partially implement a RSA PEM parser. Let's stick to the `nil` and return an error that fits the golang error description:
```
if block == nil {
  return "", errors.New("no PEM data found")
}
```

As per godoc:
```
$ go doc pem.Decode
package pem // import "encoding/pem"

func Decode(data []byte) (p *Block, rest []byte)
    Decode will find the next PEM formatted block (certificate, private key etc)
    in the input. It returns that block and the remainder of the input. If no
    PEM data is found, p is nil and the whole of the input is returned in rest.
```